File: ruby/condicionales.rb
# Definir una constante para Pi
PI = 3.1416

# Solicitar al usuario seleccionar la figura para calcular su área
# chomp se usa para eliminar el salto de línea adicional #
puts "Seleccione la figura a calcular su área:"
puts "1 para rombo"
puts "2 para rectángulo"
puts "3 para cuadrado"
puts "4 para trapecio"
print ": "
figura = gets.chomp

# Evaluar la figura seleccionada
# to_i convierte en un número entero #
case figura
when '1'  # Rombo
  print "Ingresa el valor de la diagonal mayor: "
  dmayor = gets.to_i
  print "Ingresa el valor de la diagonal menor: "
  dmenor = gets.to_i
  area = (dmayor * dmenor) / 2
  puts "El área del rombo es: #{area}"

when '2'  # Rectángulo
  print "Ingresa el valor del largo del rectángulo: "
  largo = gets.to_i
  print "Ingresa el valor del ancho del rectángulo: "
  ancho = gets.to_i
  area = largo * ancho
  puts "El área del rectángulo es: #{area}"

when '3'  # Cuadrado
  print "Ingresa el valor del lado del cuadrado: "
  lado = gets.to_i
  area = lado * lado
  puts "El área del cuadrado es: #{area}"

when '4'  # Trapecio
  print "Ingresa el valor de la base mayor: "
  bmayor = gets.to_i
  print "Ingresa el valor de la base menor: "
  bmenor = gets.to_i
  print "Ingresa la altura del trapecio: "
  altura = gets.to_i
  area = ((bmayor + bmenor) * altura) / 2
  puts "El área del trapecio es: #{area}"

else
  puts "Opción no válida."
end

File: php/condicionales.php
<?php
$Pi = 3.1416;
$const = 2;

echo "Seleccione la figura a calcular su área: \n\n 1 para rombo\n 2 para rectángulo\n 3 para cuadrado\n 4 para trapecio\n: ";
$Figura = trim(fgets(STDIN));

switch ($Figura) {
    case '1':
        echo "Ingresa el valor de la diagonal mayor: ";
        $Dmayor = (int)trim(fgets(STDIN));
        echo "Ingresa el valor de la diagonal menor: ";
        $Dmenor = (int)trim(fgets(STDIN));
        $Area = ($Dmayor * $Dmenor) / $const;
        echo "El área del rombo es: " . $Area . "\n";
        break;
    
    case '2':
        echo "Ingresa el valor del largo del rectángulo: ";
        $Largo = (int)trim(fgets(STDIN));
        echo "Ingresa el valor del ancho del rectángulo: ";
        $Ancho = (int)trim(fgets(STDIN));
        $Area = $Largo * $Ancho;
        echo "El área del rectángulo es: " . $Area . "\n";
        break;
    
    case '3':
        echo "Ingresa el valor del lado del cuadrado: ";
        $Lado = (int)trim(fgets(STDIN));
        $Area = $Lado * $Lado;
        echo "El área del cuadrado es: " . $Area . "\n";
        break;
    
    case '4':
        echo "Ingresa el valor de la base mayor: ";
        $Bmayor = (int)trim(fgets(STDIN));
        echo "Ingresa el valor de la base menor: ";
        $Bmenor = (int)trim(fgets(STDIN));
        echo "Ingresa la altura del trapecio: ";
        $H = (int)trim(fgets(STDIN));
        $Area = (($Bmayor + $Bmenor) * $H) / 2;
        echo "El área del trapecio es: " . $Area . "\n";
        break;
     
}
?>
